compiler/optimizer/RematTools.cpp

```diff
@@ -316,3 +323,25 @@ void RematTools::walkTreeTopsCalculatingRematFailureAlternatives(TR::Compilation
       start = start->getNextTreeTop();
       }
    }
+
+/*
+ * This method is used to determine if indirect loads are safe for the nodes between
+ * the start inclusive and the end exclusive. It scans for indirect stores and
+ * determines the loads to be safe if none are found. The result can be passed to
+ * gatherNodesToCheck to allow remat of stores dependent on indirect loads.
+ */
+bool RematTools::checkForSafeIndirectLoads(TR::Compilation *comp, TR::TreeTop *start,
```

Comment:
agreed - a check using the aliasing sets like done in the safety walk is probably the best idea.